### PR TITLE
No implicit any on callbacks

### LINKED ISSUE #41271: Migrate types so that `useCallback` functions are typed

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/41271

### Description
Update types in preparation for a change to the `useCallback` type in React 18. This PR should have minimal impact to actual functionality, and largely just provide more specific typing to `useCallback` handlers

### How to verify
Geen CI

### CODE REVIEW

File: frontend/src/metabase/actions/components/ActionViz/ActionParameterMapping.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const ActionParameterMappingForm = ({
   }, [action]);
 
   const handleParameterChange = useCallback(
-    (dashboardParameterId, target) => {
+    (dashboardParameterId: any, target: ParameterTarget) => {
```
Comment (npfitz) [inline frontend/src/metabase/actions/components/ActionViz/ActionParameterMapping.tsx:64]: The component that we pass `handleParameterChange` to calls for the first value to be `string | null`, and it explicitly passes null back under a couple of situations (ask the user or default value). However, `setParameterMapping()` does not expect null values, and updating that requires updating a number of functions that it passes values to. I'm not familiar enough with this area of the codebase to be confident in these changes, so I'm setting this to `any` for the time being

File: frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx [hunk 2]
```diff
@@ -21,9 +22,14 @@ export const CreateApiKeyModal = ({ onClose }: { onClose: () => void }) => {
   const secretKey = response?.data?.unmasked_key || "";
 
   const handleSubmit = useCallback(
-    async vals => {
-      await createApiKey(vals);
-      setModal("secretKey");
+    async (vals: { group_id: number | null; name: string }) => {
+      // bad
```
Comment (sloansparger) [inline frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx:26]: idk what to say about this comment... 🤨

File: frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx [hunk 3]
```diff
@@ -21,9 +22,14 @@ export const CreateApiKeyModal = ({ onClose }: { onClose: () => void }) => {
...
+      if (vals.group_id !== null) {
+        await createApiKey({
+          ...vals,
+        } as CreateApiKeyRequest);
```
Comment (sloansparger) [inline frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx:30]: why spreading here? if it's for type reasons you can sometimes do `vals as unknown as CreateApiKeyRequest`

**Review by sloansparger — APPROVED** [review]
approved w/ comments

File: frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx (hunk 2 above)
Comment (npfitz) [reply frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx:26]: Haha, I used that as markers for the `any` keys. I must have forgot to remove the comment 😂

File: frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx (hunk 3 above)
Comment (npfitz) [reply frontend/src/metabase/admin/settings/components/ApiKeys/CreateApiKeyModal.tsx:30]: You're right. This was from me trying  a couple things, but spreading isn't needed. Will fix

**Pull request merged**